pkg/traceql/ast_conditions.go

```diff
@@ -2,6 +2,23 @@ package traceql
 
 func (f SpansetFilter) extractConditions(request *FetchSpansRequest) {
 	f.Expression.extractConditions(request)
+
+	// For empty spansets { } ensure there is something that matches all spans.
+	// Use duration which would have been selected as part of the second pass
+	// metadata.
+	if s, ok := f.Expression.(Static); ok && s.Type == TypeBoolean && s.B {
```

Comment:
"kind" is the column we scan when we just want to count spans. thoughts on using kind over duration?

there's weird edge cases where it would be better:
```
{} | { <needle in the haystack conditions> }
```

but obviously that's a silly query. is the rationale that any "sane" query with `{}` will return all spans? so just pull a metadata column we know we are going to get anyway?

final Q: do you think this is the longterm fix or we'd eventually want an Op that just force pulls all spans and let the fetch layer figure it out?

Reply:
I tried kind first.  The main drawback it shows up in the results unexpectedly. It was very noticeable in the UI and didn't seem right.   Not returning it also seemed like it would be tricky, the engine would have to keep track of when it was substituted, but not also not remove it was explicitly queried by another expression.

> want an Op that just force pulls all spans
I think the direction is to try to reduce specialty in the storage layer.  The recent changes to call out the metadata fields in the second pass is a good example of that.  Would like to see more of that, so initially I would say no to a special Op to return all spans.